Language support

Oxydeas opened the pull request:
Added language translations for ENG/FR/PT/ES/IT. Fixed themes not working because of a function in util

Oxydeas commented on app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:
```diff
@@ -1,18 +1,34 @@
+        Locale locale = new Locale(getCurrentLanguage());
+        Locale.setDefault(locale);
+        config.setLocale(locale);
+        getBaseContext().getResources().updateConfiguration(config, getBaseContext().getResources().getDisplayMetrics());
```
This function is deprecated. I found no replacement after several hours

Defibrilator commented on app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:
```diff
@@ -1,18 +1,34 @@
+        Configuration config = getBaseContext().getResources().getConfiguration();
+        Locale locale = new Locale(getCurrentLanguage());
+        Locale.setDefault(locale);
+        config.setLocale(locale);
```
Question to understand better, here you get the language from the shared prefs l.24, which returns english as default, but if someone has their phone in italian let's say, does it automatically change to italian and if yes where does it do that or does the person chose in the settings menu ?

Defibrilator replied:
okay I saw your second sprint task which responds to my question : the user choses in the settings.

Oxydeas replied:
Yes, we are basically fighting the system. This code is executed in every activity and overrides the default language of the phone

Defibrilator reviewed (approved):
Thanks for your work, only one question for understanding.

aindreias commented on app/src/main/res/values-it-rIT/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="duplicate_user">Utente già presente!</string>
+    <string name="not_owner">Non sei il proprietario!</string>
+    <string name="owner_change_success">Il proprietario è stato cambiato con successo</string>
+    <string name="add_user_success">Aggiunto con successo l\'utente</string>
```
You could say "Utente aggiunto" (line 40) but we can do language edits later & this is definitely just nitpicking

Oxydeas replied:
True, my italian is rusty! I will push a fix later today, thanks for the catch. By the way @sansive if you find anything strange is spanish tell me!

aindreias reviewed (approved):
Great job and thanks for fixing the preferences!

Arkadyus reviewed (changes requested):
There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : 
![image](https://user-images.githubusercontent.com/50270465/168797293-69a7b89e-bfb9-4f0a-b93b-f63ba10d6d89.png)
Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug)
When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access
Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?

aindreias commented:
> There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : ![image](https://user-images.githubusercontent.com/50270465/168797293-69a7b89e-bfb9-4f0a-b93b-f63ba10d6d89.png) Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug) When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?

The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well.

Arkadyus commented:
> > There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : ![image](https://user-images.githubusercontent.com/50270465/168797293-69a7b89e-bfb9-4f0a-b93b-f63ba10d6d89.png) Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug) When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?
> 
> The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.

I see, sorry for that part then, ill approve the changes once the language fix is done

aindreias commented:
> > The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.
> 
> I see, sorry for that part then, ill approve the changes once the language fix is done

No problem!

Oxydeas commented:
I will investigate why you stay on mainscreen when changing languages but to be honest my actual code is only a couple of lines to change the language so I have no idea where to even begin. I will update tonight

sansive commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="debt">Deuda</string>
+    <string name="color_theme">Tema de colores</string>
+    <string name="lang_title">Idioma</string>
+    <string name="password_constraints">"- Debe contener 8 caracteres mínimos - Debe contener una mayúscula y una minúscula - Debe contener un carácter especial - No debe contener espacio - Las dos contraseñas deben ser idénticas "</string>
```
"- Debe contener mínimo 8 caracteres - Debe contener una mayúscula y una minúscula - Debe contener un carácter especial - No debe contener espacio - Las dos contraseñas deben ser idénticas "

Oxydeas replied:
Done

sansive reviewed (commented):
Spanish check done! Good work!

Arkadyus reviewed (approved):
Thanks for the fix !

Oxydeas commented:
> > > The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.
> > 
> > 
> > I see, sorry for that part then, ill approve the changes once the language fix is done
> 
> No problem!

Fixed the refresh issue by sending user to the mainscreen if theme or language changes. Fixed navbar problem

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="edit_household_notes">Editar la descripción</string>
+    <string name="edit_notes_button">"Editar "</string>
+    <string name="info_header">Residentes</string>
+    <string name="add_household">"Criar una casa "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="edit_notes_button">"Editar "</string>
+    <string name="info_header">Residentes</string>
+    <string name="add_household">"Criar una casa "</string>
+    <string name="add_join_household">"Adjuntar/criar una casa "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="discoverButton">"Descubrir "</string>
+    <string name="appDescription">\"Mejor aplicacón de organización domestica\"- los desarolladores</string>
+    <string name="settings_speakable_text">Configuración</string>
+    <string name="house_speakable_text">Seleccionar</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="appDescription">\"Mejor aplicacón de organización domestica\"- los desarolladores</string>
+    <string name="settings_speakable_text">Configuración</string>
+    <string name="house_speakable_text">Seleccionar</string>
+    <string name="info_speakable_text">"Informaciones "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="no_household_info">Aún no hay información</string>
+    <string name="edit_household_notes">Editar la descripción</string>
+    <string name="edit_notes_button">"Editar "</string>
+    <string name="info_header">Residentes</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="longitude_hint">Longitud</string>
+    <string name="leave_house">Salir</string>
+    <string name="house_name">Nombre de la casa</string>
+    <string name="permission_denied">Esta aplicación necessita este permiso para funcionar correctamente</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="remove_user_success">Borró el usuario con éxito</string>
+    <string name="QR_invalid">Este QR no es una casa válida</string>
+    <string name="remove_user">Borrar el usuario</string>
+    <string name="cant_remove_yourself">No podéis retiraros</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="QR_invalid">Este QR no es una casa válida</string>
+    <string name="remove_user">Borrar el usuario</string>
+    <string name="cant_remove_yourself">No podéis retiraros</string>
+    <string name="cant_remove_owner">El proprietario no se puede retirar</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="signature_title">Su nombre</string>
+    <string name="dark_mode_title">Modo oscuro</string>
+    <string name="attachment_title">Descargar las piezas adjuntas</string>
+    <string name="attachment_summary_on">"Descargar automáticamente las piezas adjuntas "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="dark_mode_title">Modo oscuro</string>
+    <string name="attachment_title">Descargar las piezas adjuntas</string>
+    <string name="attachment_summary_on">"Descargar automáticamente las piezas adjuntas "</string>
+    <string name="attachment_summary_off">Descaregar manualmente las piezas adjuntas</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="tap_to_modify">Tocar para modificar</string>
+    <string name="add_remove_assignee">"Añadir/retirar un asignado "</string>
+    <string name="assignees_button">Asignados</string>
+    <string name="no_events">No hay eventos ese día</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="confirm">"Confirmar "</string>
+    <string name="email_already_used">Correo electrónico ya utilizado</string>
+    <string name="verify_email">"Comprobe su dirección de correo por favor "</string>
+    <string name="sign_out">Desconexión</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="event_editing_title">Editar evento</string>
+    <string name="confirm">"Confirmar "</string>
+    <string name="email_already_used">Correo electrónico ya utilizado</string>
+    <string name="verify_email">"Comprobe su dirección de correo por favor "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="email_already_used">Correo electrónico ya utilizado</string>
+    <string name="verify_email">"Comprobe su dirección de correo por favor "</string>
+    <string name="sign_out">Desconexión</string>
+    <string name="sign_in_with_email">Conexión por correo electrónico</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="sign_out">Desconexión</string>
+    <string name="sign_in_with_email">Conexión por correo electrónico</string>
+    <string name="sign_up">"Crear una cuenta "</string>
+    <string name="enter_email">Introducir correo electrónico</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="sign_in_with_email">Conexión por correo electrónico</string>
+    <string name="sign_up">"Crear una cuenta "</string>
+    <string name="enter_email">Introducir correo electrónico</string>
+    <string name="enter_password">Entrar contraseña</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="email_not_valid">Correo electrónico inválido</string>
+    <string name="password_not_valid">Contraseña inválida</string>
+    <string name="inputs_empty">Los campos no pueden estar vacíos</string>
+    <string name="signout_intent">"Desconexión "</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="show">Mostrar</string>
+    <string name="remove">Suprimir</string>
+    <string name="three_dots">Tres puntos</string>
+    <string name="event_attachment_content_description">Los accesorios para el evento</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="three_dots">Tres puntos</string>
+    <string name="event_attachment_content_description">Los accesorios para el evento</string>
+    <string name="name">Nombre</string>
+    <string name="add_item_title">Añadir un elemento à la lista</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="event_attachment_content_description">Los accesorios para el evento</string>
+    <string name="name">Nombre</string>
+    <string name="add_item_title">Añadir un elemento à la lista</string>
+    <string name="item_delete_button_description">Suprimir el elemento correspondiente</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="name">Nombre</string>
+    <string name="add_item_title">Añadir un elemento à la lista</string>
+    <string name="item_delete_button_description">Suprimir el elemento correspondiente</string>
+    <string name="add_item_button">Añadir un elemento</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="add_item_title">Añadir un elemento à la lista</string>
+    <string name="item_delete_button_description">Suprimir el elemento correspondiente</string>
+    <string name="add_item_button">Añadir un elemento</string>
+    <string name="remove_picked_up">Retirar los elementos recuperados</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="offline_warning">"Se ha desconectado. "</string>
+    <string name="show_qr_code">Mostrar código QR</string>
+    <string name="show_invite_qr_code">Mostrar código QR de invitación</string>
+    <string name="house_location_title">Localisación de la casa</string>
```
Done

Oxydeas commented on app/src/main/res/values-es-rES/strings.xml:
```diff
@@ -0,0 +1,133 @@
+    <string name="text_location_permission">Necesitamos acceso a su ubicación para acceder a la casa más cercana</string>
+    <string name="go_back_online_button">Volver a conectarse</string>
+    <string name="go_offline_button">Desconectarse</string>
+    <string name="address_field">"Introducir dirección "</string>
```
Done

aindreias reviewed (approved):
Thanks for the changes Luca, and for the theme-related fixes as well. Great work!

sansive reviewed (approved):
Thanks for the translations!
